src/main/java/org/opensearch/security/ssl/transport/SecuritySSLRequestHandler.java

```diff
@@ -86,6 +89,11 @@ public final void messageReceived(T request, TransportChannel channel, Task task
 
         ThreadContext threadContext = getThreadContext();
 
+        String channelType = channel.getChannelType();
+        if (!DEFAULT_CHANNEL_TYPES.contains(channelType)) {
+            channel = getInnerChannel(channel);
+        }
```

Comment:
Nit - Ideally it is good practice to avoid mutation of method reference, but I can see that creating a new reference can create more confusion.